hypershift-operator/controllers/sharedingress/sharedingress_controller.go

```diff
@@ -34,9 +38,20 @@ func (r *SharedIngressReconciler) SetupWithManager(mgr ctrl.Manager, createOrUpd
 	mgr.GetCache().IndexField(context.Background(), &corev1.Service{}, "metadata.name", func(o client.Object) []string {
 		return []string{o.GetName()}
 	})
+
+	// A channel is used to generate an initial sync event.
+	// Afterwards, the controller syncs on HostedClusters.
+	initialSync := make(chan event.GenericEvent)
+
 	builder := ctrl.NewControllerManagedBy(mgr).
 		For(&hyperv1.HostedCluster{}).
+		WatchesRawSource(source.Channel(initialSync, &handler.EnqueueRequestForObject{})).
 		Named("SharedIngressController")
+
+	go func() {
+		initialSync <- event.GenericEvent{Object: &hyperv1.HostedCluster{}}
+	}()
```

Comment:
wouldn't you need a defer?

Reply:
no its not needed, this is an unbuffered channel, so the send will automatically for a receiver.